### PR TITLE
Reader compilation refactor

### PR DESCRIPTION
**Pull request by andyfengHKU**

### CODE REVIEW

File: src/processor/map/map_copy_from.cpp
Comment (ray6080) [inline src/processor/map/map_copy_from.cpp:1]: Move `mapCopyRel` to this file too.

File: src/processor/map/map_copy.cpp
Comment (ray6080) [inline src/processor/map/map_copy.cpp:1]: rename this file to `map_copy_to.cpp`.

File: src/include/processor/operator/persistent/reader.h [hunk 1]
```diff
@@ -40,13 +50,18 @@ class Reader : public PhysicalOperator {
...
 private:
-    ReaderInfo readerInfo;
+    std::unique_ptr<ReaderInfo> info;
     std::shared_ptr<storage::ReaderSharedState> sharedState;
-    std::unique_ptr<common::DataChunk> dataChunkToRead;
+
     storage::LeftArrowArrays leftArrowArrays;
 
+    std::unique_ptr<common::DataChunk> dataChunk = nullptr;
+    common::ValueVector* nodeOffsetVector = nullptr;
```
Comment (ray6080) [inline src/include/processor/operator/persistent/reader.h:59]: Do we need this field `nodeOffsetVector`? I would just fetch it dynamically from `resultSet`.

File: src/include/processor/operator/persistent/reader.h [hunk 2]
```diff
@@ -40,13 +50,18 @@ class Reader : public PhysicalOperator {
...
 
 private:
-    ReaderInfo readerInfo;
+    std::unique_ptr<ReaderInfo> info;
     std::shared_ptr<storage::ReaderSharedState> sharedState;
-    std::unique_ptr<common::DataChunk> dataChunkToRead;
+
     storage::LeftArrowArrays leftArrowArrays;
 
+    std::unique_ptr<common::DataChunk> dataChunk = nullptr;
```
Comment (ray6080) [inline src/include/processor/operator/persistent/reader.h:58]: Maybe move all initializations to constructor. Right now, i see `readFuncData` is initialized in the constructor, while others are here. Let's unify them.

File: src/include/planner/logical_plan/copy/logical_copy_from.h [hunk 3]
```diff
@@ -10,53 +10,45 @@ namespace planner {
...
     inline std::shared_ptr<binder::Expression> getNodeOffsetExpression() const {
         return nodeOffsetExpression;
     }
-
     inline std::shared_ptr<binder::Expression> getOutputExpression() const {
         return outputExpression;
     }
 
-    inline bool isPreservingOrder() { return preservingOrder; }
+    inline bool preservingOrder() { return preservingOrder_; }
```
Comment (ray6080) [inline src/include/planner/logical_plan/copy/logical_copy_from.h:35]: Consider renaming to `orderPreserving` or `preserveOrder`, ditto on the field `preservingOrder_`.

File: src/include/binder/copy/bound_copy_from.h [hunk 4]
```diff
@@ -13,22 +13,30 @@ namespace binder {
...
 
 private:
-    common::CopyDescription copyDescription;
-    common::table_id_t tableID;
-    std::string tableName;
+    std::unique_ptr<common::CopyDescription> copyDescription;
+    catalog::TableSchema* tableSchema;
+    expression_vector columnExpressions;
+    std::shared_ptr<Expression> nodeOffsetExpression;
+    bool readInSerialMode_;
```
Comment (ray6080) [inline src/include/binder/copy/bound_copy_from.h:39]: I would rename this to  be consistent with `LogicalCopyFrom` as `preservingOrder_`.

**Review by ray6080 — APPROVED** [review]

File: src/include/processor/operator/persistent/reader.h (hunk 1 above)
Comment (andyfengHKU) [reply src/include/processor/operator/persistent/reader.h:59]: This is to be consistent across our operator implementation

**Pull request merged**